types/context.go

```diff
@@ -302,6 +302,22 @@ func (c Context) CacheContext() (cc Context, writeCache func()) {
 	return cc, writeCache
 }
 
+// CloneMultiStore returns a new Context with a cloned cached multi-store.
+func (c Context) CloneMultiStore() Context {
+	return c.WithMultiStore(c.ms.Clone())
+}
```

Comment:
This PR replaces several calls to CacheContext with calls to CloneMultiStore. However, these methods are not equivalent in behavior. Specifically, CacheContext calls c.ms.CacheMultiStore, and CloneMultiStore does not; that former method does several things which impact consumer use.

Reply:
you are right, there's event manager to handle, I'll see what can do.

Reply:
Removed the whole `CloneMultiStore` method, and make the `RunAtomic` method to handle event correctly, one less method to worry about.